更新最新测试版后部分书源不能用了

### 确认 / Assignments

- [X] 搜索现有issues，不存在相似或相关的issue / No similar or related issues
- [X] 最新[测试版](https://github.com/gedoor/legado/actions/workflows/test.yml)依然存在此问题 / Latest beta app does not work
- [X] 此问题和Xposed、Lsposed、Magisk、手机主题、浏览器插件等无关 / Make sure your machine is not touched by hook frameworks, plugins etc

### 问题描述 / Describe Bugs

最新测试版升级rhino后部分书源不能用了

### 复现步骤 / How to reproduce

最新测试版升级rhino后部分书源不能用了，之前版本都是正常
这是书源调试时的日志
[00:10.417] com.script.ScriptException: org.mozilla.javascript.EcmaError: TypeError: Cannot read property "1" from null (<Unknown source>#5) in <Unknown source> at line number 5
	at com.script.rhino.RhinoScriptEngine.eval(SourceFile:33)
	at com.script.AbstractScriptEngine.eval(SourceFile:3)
	at io.legado.app.model.analyzeRule.AnalyzeRule.evalJS(SourceFile:119)
	at io.legado.app.model.analyzeRule.AnalyzeRule.getString(SourceFile:32)
	at io.legado.app.model.analyzeRule.AnalyzeRule.getString$default(SourceFile:2)
	at io.legado.app.model.analyzeRule.AnalyzeRule.getString(SourceFile:11)
	at io.legado.app.model.webBook.w.a(SourceFile:8)
	at io.legado.app.model.webBook.w.b(SourceFile:34)
	at io.legado.app.model.webBook.x0.k(SourceFile:463)
	at io.legado.app.model.webBook.s0.invokeSuspend(SourceFile)
	at m7.a.resumeWith(SourceFile:9)
	at kotlinx.coroutines.m0.run(SourceFile:127)
	at com.bumptech.glide.load.engine.a.run(SourceFile:15)
	at f9.k.run(SourceFile:3)
	at f9.b.run(SourceFile:96)
Caused by: org.mozilla.javascript.EcmaError: TypeError: Cannot read property "1" from null (<Unknown source>#5)
	at org.mozilla.javascript.ScriptRuntime.constructError(SourceFile:9)
	at org.mozilla.javascript.ScriptRuntime.constructError(SourceFile:3)
	at org.mozilla.javascript.ScriptRuntime.typeError(SourceFile:3)
	at org.mozilla.javascript.ScriptRuntime.typeErrorById(SourceFile:5)
	at org.mozilla.javascript.ScriptRuntime.undefReadError(SourceFile:20)
	at org.mozilla.javascript.ScriptRuntime.getObjectIndex(SourceFile:6)
	at org.mozilla.javascript.Interpreter.doGetElem(SourceFile:30)
	at org.mozilla.javascript.Interpreter.interpretLoop(SourceFile:99)
	at org.mozilla.javascript.Interpreter.interpret(SourceFile:57)
	at org.mozilla.javascript.InterpretedFunction.call(SourceFile:21)
	at org.mozilla.javascript.ContextFactory.doTopCall(SourceFile:1)
	at com.script.rhino.RhinoScriptEngine$1.superDoTopCall(SourceFile:1)
	at com.script.rhino.RhinoScriptEngine$1.doTopCall(SourceFile:60)
	at org.mozilla.javascript.ScriptRuntime.doTopCall(SourceFile:8)
	at org.mozilla.javascript.InterpretedFunction.exec(SourceFile:23)
	at org.mozilla.javascript.Context.evaluateReader(SourceFile:7)
	at com.script.rhino.RhinoScriptEngine.eval(SourceFile:25)
	... 14 more


### 确认 / Assignment

- [X] 已经提交复现所需要的附加资料 / Submit additions related with bugs

### 其他信息 / Additions

_No response_

### 日志提交 / Relevant log output

_No response_

### 阅读版本 / Legado version

3.24.050213

### Android版本 / Android version

6

### 机型 / Model

sm919

--- a/modules/rhino/src/main/java/com/script/rhino/RhinoScriptEngine.kt
+++ b/modules/rhino/src/main/java/com/script/rhino/RhinoScriptEngine.kt
@@ -310,7 +310,7 @@ object RhinoScriptEngine : AbstractScriptEngine(), Invocable, Compilable {
             override fun hasFeature(cx: Context, featureIndex: Int): Boolean {
                 @Suppress("UNUSED_EXPRESSION")
                 return when (featureIndex) {
-                    //Context.FEATURE_ENABLE_JAVA_MAP_ACCESS -> true
+                    Context.FEATURE_ENABLE_JAVA_MAP_ACCESS -> true
                     else -> super.hasFeature(cx, featureIndex)
                 }
             }

